app/src/main/java/ch/sdp/vibester/activity/profile/ProfileActivity.kt

```diff
@@ -1,204 +1,226 @@
-
-        if (user.uid != "") {
-            generateQrCode(user.uid)
-        }
-    }
-
-    /**
-     * generate the qr code bitmap of the given data
-     * @param data Qr Code data
-     */
-    private fun generateQrCode(data: String) {
-        val size = 512
-        val hints = HashMap<EncodeHintType?, Any?>()
-        hints[EncodeHintType.ERROR_CORRECTION] = ErrorCorrectionLevel.H
-
-        val bits = QRCodeWriter().encode(data, BarcodeFormat.QR_CODE, size, size)
-        val bmp = MatrixToImageWriter.toBitmap(bits)
-        val qrCodeCanvas = Canvas(bmp)
-
-        val scaleFactor = 4 // resize the image
-        val logo = BitmapFactory.decodeStream(assets.open("logo.png"))
-        logo.density = logo.density * scaleFactor
-        val xLogo = (size - logo.width / scaleFactor) / 2f
-        val yLogo = (size - logo.height / scaleFactor) / 2f
-
-        qrCodeCanvas.drawBitmap(logo, xLogo, yLogo, null)
-
-        findViewById<ImageView>(R.id.qrCode).setImageBitmap(bmp)
-    }
-}
-
+package ch.sdp.vibester.activity.profile
+
+import android.app.Activity
+import android.content.Intent
+import android.graphics.*
+import android.net.Uri
+import android.os.Bundle
+import android.view.View.*
+import android.view.Window
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.NestedScrollView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.sdp.vibester.R
+import ch.sdp.vibester.activity.MainActivity
+import ch.sdp.vibester.auth.FireBaseAuthenticator
+import ch.sdp.vibester.database.DataGetter
+import ch.sdp.vibester.database.ImageGetter
+import ch.sdp.vibester.helper.ImageHelper
+import ch.sdp.vibester.helper.IntentSwitcher
+import ch.sdp.vibester.user.OnItemClickListener
+import ch.sdp.vibester.user.ProfileFollowingAdapter
+import ch.sdp.vibester.user.User
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.qrcode.QRCodeWriter
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import dagger.hilt.android.AndroidEntryPoint
+import net.glxn.qrgen.android.MatrixToImageWriter
+import javax.inject.Inject
+
+/**
+ * Display user profile's data (image, username, scores, etc.) in UI
+ */
+@AndroidEntryPoint
+open class ProfileActivity : AppCompatActivity(), OnItemClickListener {
+    @Inject
+    lateinit var dataGetter: DataGetter
+
+    @Inject
+    lateinit var authenticator: FireBaseAuthenticator
+
+    @Inject
+    lateinit var imageGetter: ImageGetter
+
+    private val imageSize = 1000
+    val imageRequestCode = 100
+
+    private var followings: MutableList<User> ? = null
+    private var profileFriendsAdapter: ProfileFollowingAdapter?= null
+
+    /**
+     * Generic onCreate method belonging to ProfileActivity.
+     */
```

Comment:
not sure this is useful to redefine Javadoc for an override 😅

Reply:
I didn't add it but I can remove it.